desktop/src/com/mygdx/game/client/ecs/system/MovementSystem.java

```diff
@@ -20,26 +23,43 @@ public class MovementSystem extends IteratingSystem {
 
   private final ChosenEntity chosenEntity;
   private final MoveEntityService moveEntityService;
+  private final OutOfMoveRangeDialogFactory moveRangeDialog;
   private ComponentMapper<Highlighted> highlightedMapper;
   private ComponentMapper<Coordinates> coordinatesMapper;
+  private ComponentMapper<MoveRange> moveRangeMapper;
   private ComponentMapper<Field> fieldMapper;
 
   @Inject
   public MovementSystem(
       @NonNull ChosenEntity chosenEntity,
-      @NonNull MoveEntityService moveEntityService
+      @NonNull MoveEntityService moveEntityService,
+      @NonNull OutOfMoveRangeDialogFactory moveRangeDialog
   ) {
     this.chosenEntity = chosenEntity;
     this.moveEntityService = moveEntityService;
+    this.moveRangeDialog = moveRangeDialog;
   }
 
   @Override
   protected void process(int entityId) {
     if (chosenEntity.isAnyChosen() && fieldMapper.has(chosenEntity.peek())) {
       log.info("some are chosen and there's a movable highlighted entity");
+
       var targetCoordinate = coordinatesMapper.get(chosenEntity.pop());
-      moveEntityService.moveEntity(entityId, targetCoordinate);
-      highlightedMapper.remove(entityId);
+      var currentCoordinate = coordinatesMapper.get(entityId);
+
+      var distance = DistanceUtil.distance(currentCoordinate, targetCoordinate);
+      var range = moveRangeMapper.get(entityId).getCurrentRange();
+
+      log.info("Chce przejsc " + distance + " ale zostało mi " + range);
+
+      if (distance > range){
```

Comment:
Tu mam wątpliwość pewną bo sprawdzamy to samo w MoveEntityService, ale chyba jest okej, bo pomaga to od razu wyświetlić wiadomość, a w serwerze jest jeszcze dodatkowe sprawdzanie, WDYT @Karkoon

Reply:
dla mnie ok